### PR TITLE
Module description label and due date skeletons

### PR DESCRIPTION
**Pull request by brissons**
## Description

When [this PR was merged](https://github.com/BrightspaceHypermediaComponents/activities/pull/1876), I noticed some skeletons not matching up with the rendered element. Specifically:

- `d2l-activity-content-module-detail`'s **Description** label spans the width of the page rather than the label.
- Due Date skeleton was only for the collapsed version, rather than the expanded version with the label.

Instead of using the placeholder skeleton, I leveraged the built-in skeletons for the due date editor (both expanded and collapsed) so that it better lines up with the rendered elements.

## Goal/Solution

We want the loading skeletons to line up with the elements they represent.

## Video/Screenshots
- Description label [before](https://user-images.githubusercontent.com/52549862/126234981-f708edeb-6daa-46e6-b357-7c38d2d04793.png) and [after](https://user-images.githubusercontent.com/52549862/126234880-e0eb3b64-dedf-4a73-9270-202afd2769df.png).
- Due date editor expanded [before](https://user-images.githubusercontent.com/52549862/126237208-0a35cd61-7e40-498b-bfa5-b97d0c9bd6e6.png) and [after](https://user-images.githubusercontent.com/52549862/126237634-908955b8-0d03-429e-b93a-631863bfbbd4.png).
- Due date editor collapsed [before](https://user-images.githubusercontent.com/52549862/126237208-0a35cd61-7e40-498b-bfa5-b97d0c9bd6e6.png) and [after](https://user-images.githubusercontent.com/52549862/126237721-90f0ff83-dbb8-4eb0-a4de-50e3c38e1bb1.png) (small change).
- Module example [before](https://user-images.githubusercontent.com/52549862/126235693-fbb54e1e-8a22-48bf-bbf4-9d14443b7a19.gif) and [after](https://user-images.githubusercontent.com/52549862/126238011-491dd0d7-e4f0-4fe5-955d-884ec24586e7.gif)
- Weblink example [before ](https://user-images.githubusercontent.com/52549862/126235535-d907f03e-1e8e-4288-b629-5f5529285bbe.gif) and [after](https://user-images.githubusercontent.com/52549862/126237876-81af3c75-0f15-484e-a5d9-58a8bd69c85f.gif).

### CODE REVIEW

**Review by ianlaidlaw — DISMISSED** [review]
code LGTM, i will dev test today!

File: components/d2l-activity-editor/d2l-activity-content-editor/shared-components/d2l-activity-content-editor-due-date.js [hunk 1]
```diff
@@ -27,47 +27,41 @@ class ContentEditorDueDate extends SkeletonMixin(LocalizeActivityEditorMixin(Rtl
...
+					margin-bottom: 18px;
 				}
 			`
 		];
 	}
 
 	constructor() {
 		super();
-		this._hasDatePermissions = false;
+		this._hasDatePermissions = true;
```
Comment (ChrisKraljevicD2L) [inline components/d2l-activity-editor/d2l-activity-content-editor/shared-components/d2l-activity-content-editor-due-date.js:38]: Why do we want this defaulted to `true`?

Comment (brissons) [reply components/d2l-activity-editor/d2l-activity-content-editor/shared-components/d2l-activity-content-editor-due-date.js:38]: This way, the skeleton for the due date editor shows up on initial page load. When the entity is resolved and if`_hasDatePermissions` is false, then the loading skeleton for the due date editor is removed before the whole page resolves (existing behavior). If we were to do it the other way, then we would have the date editor skeleton pop-in while the page is being loaded.

Having said that, your question raises a good point: how do we want to tackle this in general? Do we want to default showing all the loading skeletons and then remove them if they don't have permission? Or have more skeletons pop-in as the entity loads?

Comment (ChrisKraljevicD2L) [reply components/d2l-activity-editor/d2l-activity-content-editor/shared-components/d2l-activity-content-editor-due-date.js:38]: Skeletons popping in seems like unwanted behaviour. If there was a way to get those skeletons to show without needing to know if date permissions are on or off that would be ideal. Taking skeletons away if permissions come back false is better than content popping in, skeletonized or not.

Comment (brissons) [reply components/d2l-activity-editor/d2l-activity-content-editor/shared-components/d2l-activity-content-editor-due-date.js:38]: Good point! addressed in https://github.com/BrightspaceHypermediaComponents/activities/pull/1897/commits/913bec2da81ee12829ee541cf5d9d729ef09fe3a.

**Review by ianlaidlaw — APPROVED** [review]

**Review by prajapac — APPROVED** [review]

**Pull request merged**